src/main/java/software/amazon/cloudformation/resource/ResourceTypeSchema.java

```diff
@@ -149,33 +148,14 @@ public static ResourceTypeSchema load(final JSONObject schemaJson) {
         return Collections.unmodifiableMap(this.unprocessedProperties);
     }
 
-    public boolean hasWriteOnlyProperties(final JSONObject resourceModel) {
-        return this.getWriteOnlyPropertiesAsStrings().stream()
-            .anyMatch(writeOnlyProperty -> hasProperty(RegExUtils.removeFirst(writeOnlyProperty, "/properties"), resourceModel));
-    }
-
-    public boolean hasProperty(final String property, final JSONObject resourceModel) {
-        return hasProperty(new JSONPointer(property), resourceModel);
-    }
-
-    public boolean hasProperty(final JSONPointer property, final JSONObject resourceModel) {
-        try {
-            return property.queryFrom(resourceModel) != null;
-        } catch (JSONPointerException e) {
-            return false;
-        }
-    }
-
     public void removeWriteOnlyProperties(final JSONObject resourceModel) {
         this.getWriteOnlyPropertiesAsStrings().stream()
-            .forEach(writeOnlyProp -> removeProperty(new JSONPointer(RegExUtils.removeFirst(writeOnlyProp, "/properties")),
+            .forEach(writeOnlyProperty -> removeProperty(new PublicJSONPointer(writeOnlyProperty.replaceFirst("/properties", "")),
```

Comment:
yeah, this works great. just as a warning, `replaceFirst` takes a regex. could also anchor it or test via `startsWith`? in fact, would probably be good if `getWriteOnlyPropertiesAsStrings` just did this by default. but can iterate in future PRs